core/trino-main/src/main/java/io/trino/sql/planner/DistributedExecutionPlanner.java

```diff
@@ -203,12 +207,21 @@ private Visitor(
                 dynamicFilter = dynamicFilterService.createDynamicFilter(session.getQueryId(), dynamicFilters, node.getAssignments(), typeProvider);
             }
 
+            ConstraintEvaluator constraintEvaluator = ConstraintEvaluator.EMPTY;
+            Optional<ConstraintEvaluator> existingEvaluator = filter.map(FilterNode::getPredicate)
+                    .filter(expression -> !DynamicFilters.isDynamicFilter(expression))
+                    .map(expression -> new LayoutConstraintEvaluator(metadata, typeAnalyzer, session, typeProvider, node.getAssignments(), expression));
+            if (existingEvaluator.isPresent()) {
+                constraintEvaluator = existingEvaluator.get();
```

Comment:
You don't need two variables.

```
constraintEvaluator = filter.map
.map
.orElse(ConstraintEvaluator.ALWAYS_TRUE)
```

Reply:
```
ConstraintEvaluator existingEvaluator = filterPredicate
                    .map(predicate -> filterConjuncts(metadata, predicate, expression -> !DynamicFilters.isDynamicFilter(expression)))
                    .map(expression -> new LayoutConstraintEvaluator(metadata, typeAnalyzer, session, typeProvider, node.getAssignments(), expression))
                    .orElse(ConstraintEvaluator. ALWAYS_TRUE);

This doesn't compile, it says `LayoutConstraintEvaluator is required` instead of `ConstraintEvaluator`. 
Type erasure I guess.

Reply:
this could be fixed with explicit casting like 
```
ConstraintEvaluator existingEvaluator = filterPredicate
                    .map(predicate -> filterConjuncts(metadata, predicate, expression -> !DynamicFilters.isDynamicFilter(expression)))
                    .map(expression -> ((ConstraintEvaluator)new LayoutConstraintEvaluator(metadata, typeAnalyzer, session, typeProvider, node.getAssignments(), expression)))
                    .orElse(ConstraintEvaluator. ALWAYS_TRUE);
```

tough I am not sure i like it

Reply:
IMO it looks better even with the cast.

Reply:
ok I will change it